pylint/lint/pylinter.py

```diff
@@ -614,31 +615,39 @@ def initialize(self) -> None:
             if not msg.may_be_emitted(self.config.py_version):
                 self._msgs_state[msg.msgid] = False
 
-    def _discover_files(self, files_or_modules: Sequence[str]) -> Iterator[str]:
+    def _discover_files(
```

Comment:
chose to modify this to keep the ability to be efficient and not re-traverse subtrees.